### PR TITLE
Azure Private IP Addresses In Dynamic Inventory

### PR DESCRIPTION
**Pull request by rossedman**
##### SUMMARY
Includes options for returning private ip addresses in dynamic inventory script for Azure

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
/contrib/inventory/azure_rm.py

##### ANSIBLE VERSION
```
ansible 2.3.1.0
  config file =
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jul 18 2017, 09:17:00) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### ADDITIONAL INFORMATION
Tested before cloud changes to inventory script. Haven't seen those. Should not effect it, but would like a second set of eyes on it. Thanks! 👍

### CODE REVIEW

**Comment by xscript** [comment]
@ZhijunZhao please take a look.

**Comment by maxlock** [comment]
👍  I also manage azure instances behind a bastion, I could do with this being added thanks

File: contrib/inventory/azure_rm.py [hunk 1]
```diff
@@ -646,11 +647,15 @@ def _load_machines(self, machines):
...
+                        if self.include_private_ip:
+                            host_vars['ansible_host'] = ip_config.private_ip_address
                         if ip_config.public_ip_address:
                             public_ip_reference = self._parse_ref_id(ip_config.public_ip_address.id)
                             public_ip_address = self._network_client.public_ip_addresses.get(
                                 public_ip_reference['resourceGroups'],
                                 public_ip_reference['publicIPAddresses'])
+                            if not self.include_private_ip:
+                                host_vars['ansible_host'] = public_ip_address.ip_address
                             host_vars['ansible_host'] = public_ip_address.ip_address
```
Comment (yuwzho) [inline contrib/inventory/azure_rm.py:659]: Sorry, I m really confused what's the logic of the `include_private_ip`. Is this 
```
ANSIBLE[HOST] = INCLUDE_PRIVATE_IP ? PRIVATE_IP : PUBLIC_IP
```
or something else?

**Review by yuwzho — CHANGES_REQUESTED** [review]

File: contrib/inventory/azure_rm.py (hunk 1 above)
Comment (rossedman) [reply contrib/inventory/azure_rm.py:659]: If private_ips are not enabled then return public ip addresses for the ansible_host. This maintains the original data that is returned.

Comment (yuwzho) [reply contrib/inventory/azure_rm.py:659]: Seems the line after the conditional check is a potential bug..

Comment (rossedman) [reply contrib/inventory/azure_rm.py:659]: I can see why you would say that. Is there a testing harness for this? Would write unit tests but seems like it needs to be refactored quite a bit to do that.

Comment (yuwzho) [reply contrib/inventory/azure_rm.py:659]: My point is from [line 657 to 659](https://github.com/rossedman/ansible/blob/b72d0a46218baf2a98df19e703ce8647b65b5d6a/contrib/inventory/azure_rm.py#L657-L659), you always set `host_vars['ansible_host'] = public_ip_address.ip_address`, no matter what the conditional check result is

Comment (rossedman) [reply contrib/inventory/azure_rm.py:659]: Oh, man. That's not supposed to be there. I see what you mean. I had run this for awhile and then ported over. Let me push update. Duh.

**Comment by yuwzho** [comment]
duplicate_of #36109

**Pull request closed without merging**